File: Monthly Challenges/September 2020 Monthly Challenge/Day26.java
public class Day26 {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        int time = 0;
        if (timeSeries.length == 0) {
            return time;
        }
        
        time = duration * timeSeries.length;
        for (int i = 1; i < timeSeries.length; ++i) {
            time += Math.min(0, timeSeries[i] - timeSeries[i - 1] - duration);
        }
        
        return time;
    }
}

File: Monthly Challenges/October 2020 Monthly Challenge/Day26.scala
object Day26 {
    final val NUM_GLASS_ROWS = 100

    def champagneTower(poured: Int, query_row: Int, query_glass: Int): Double = {
        val glass_matrix = Array.ofDim[Double](NUM_GLASS_ROWS + 1, NUM_GLASS_ROWS + 1)
        glass_matrix(0)(0) = poured.toDouble
        math.min(fillGlasses(glass_matrix)(query_row)(query_glass), 1)
    }

    def fillGlasses(glass_matrix: Array[Array[Double]]): Array[Array[Double]] = {
        for (row <- 0 until glass_matrix.length - 1 ; idx <- 0 until glass_matrix(0).length - 1) {
            val flood_quantity = (glass_matrix(row)(idx) - 1) / 2
            if (flood_quantity > 0) {
                glass_matrix(row + 1)(idx) += flood_quantity
                glass_matrix(row + 1)(idx + 1) += flood_quantity
            }
        }
        glass_matrix
    }
}
